components/script/dom/htmlimageelement.rs

```diff
@@ -1891,164 +1891,143 @@ pub fn collect_sequence_characters<F>(s: &str, predicate: F) -> (&str, &str)
-        if !predicate(&ch) {
-            return (&s[0..i], &s[i..]);
-        }
-    }
-
-    (s, "")
+    let i = s.find(|ch| !predicate(&ch)).unwrap_or(s.len());
+    (&s[0..i], &s[i..])
 }
 
 /// Parse an `srcset` attribute:
 /// <https://html.spec.whatwg.org/multipage/#parsing-a-srcset-attribute>.
 pub fn parse_a_srcset_attribute(input: &str) -> Vec<ImageSource> {
-    let mut url_len = 0;
-    let mut candidates: Vec<ImageSource> = vec![];
-    while url_len < input.len() {
-        let position = &input[url_len..];
+    let mut input_pos = 0;
+    let mut candidates = vec![];
+    while input_pos < input.len() {
+        let position = &input[input_pos..];
         let (spaces, position) =
-            collect_sequence_characters(position, |c| *c == ',' || char::is_whitespace(*c));
+            collect_sequence_characters(position, |c| *c == ',' || char::is_ascii_whitespace(c));
         // add the length of the url that we parse to advance the start index
-        let space_len = spaces.char_indices().count();
-        url_len += space_len;
+        input_pos += spaces.len();
         if position.is_empty() {
             return candidates;
         }
-        let (url, spaces) = collect_sequence_characters(position, |c| !char::is_whitespace(*c));
+        let (url, position) =
+            collect_sequence_characters(position, |c| !char::is_ascii_whitespace(c));
         // add the counts of urls that we parse to advance the start index
-        url_len += url.chars().count();
-        let comma_count = url.chars().rev().take_while(|c| *c == ',').count();
-        let url: String = url
-            .chars()
-            .take(url.chars().count() - comma_count)
-            .collect();
-        // add 1 to start index, for the comma
-        url_len += comma_count + 1;
-        let (space, position) = collect_sequence_characters(spaces, |c| char::is_whitespace(*c));
-        let space_len = space.len();
-        url_len += space_len;
+        input_pos += url.len();
+
+        let trimmed_url = url.trim_end_matches(',');
         let mut descriptors = Vec::new();
-        let mut current_descriptor = String::new();
-        let mut state = ParseState::InDescriptor;
-        let mut char_stream = position.chars().enumerate();
-        let mut buffered: Option<(usize, char)> = None;
-        loop {
-            let next_char = buffered.take().or_else(|| char_stream.next());
-            if next_char.is_some() {
-                url_len += 1;
-            }
-            match state {
-                ParseState::InDescriptor => match next_char {
-                    Some((_, ' ')) => {
-                        if !current_descriptor.is_empty() {
-                            descriptors.push(current_descriptor.clone());
-                            current_descriptor = String::new();
-                            state = ParseState::AfterDescriptor;
-                        }
-                        continue;
```

Comment:
I think this is the bug with '/n'. We need char::is_ascii_whitespace check like in new version.